src/pages/SettingsPage.tsx

```diff
@@ -46,42 +46,47 @@ export const SettingsPage = () => {
   const [showDeleteAccountDialog, setShowDeleteAccountDialog] = useState(false)
   const [isDeleting, setIsDeleting] = useState(false)
   const { user } = useAuth()
+  const storeLicense = useLicenseStore((state) => state.license)
+  const isLicenseLoading = useLicenseStore((state) => state.isLoading)
+
+  const hasProLicense = !isLicenseLoading && storeLicense && (storeLicense.status === 'active' || storeLicense.status === 'trialing')
+  const maxDevicesToShow = hasProLicense ? 3 : 1
```

Comment:
Might want to put this "maxDevicesToShow" into something shared as well since we are doing this in more than one place